servicetalk-encoding-netty/src/test/java/io/servicetalk/encoding/netty/NettyCompressionStreamingSerializerTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2022 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.encoding.netty;
+
+import io.servicetalk.buffer.api.Buffer;
+import io.servicetalk.buffer.api.BufferAllocator;
+import io.servicetalk.buffer.api.CompositeBuffer;
+import io.servicetalk.concurrent.api.Publisher;
+import io.servicetalk.serializer.api.StreamingSerializerDeserializer;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+import static io.servicetalk.buffer.netty.BufferAllocators.DEFAULT_ALLOCATOR;
+import static io.servicetalk.concurrent.api.Publisher.from;
+import static io.servicetalk.encoding.netty.NettyCompression.deflateDefaultStreaming;
+import static io.servicetalk.encoding.netty.NettyCompression.gzipDefaultStreaming;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+
+class NettyCompressionStreamingSerializerTest {
+    enum StreamingType {
+        GZIP(gzipDefaultStreaming()), DEFLATE(deflateDefaultStreaming());
+
+        final StreamingSerializerDeserializer<Buffer> serializer;
+
+        StreamingType(StreamingSerializerDeserializer<Buffer> serializer) {
+            this.serializer = serializer;
+        }
+    }
+
+    @ParameterizedTest(name = "{index}: flushStrategy = {0}")
```

Comment:
flushStrategy -> algorithm?

Reply:
good catch. I'll change to map to parameter name `type`.